contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -527,6 +527,47 @@ CAPIStatus Filter::continueStatus(ProcessorState& state, GolangStatus status) {
   return CAPIStatus::CAPIOK;
 }
 
+CAPIStatus Filter::addData(ProcessorState& state, absl::string_view data, bool is_streaming) {
+  if (state.filterState() == FilterState::ProcessingData) {
+    // Calling add{Decoded,Encoded}Data when processing data will mess up the buffer management
+    // in Golang filter. And more importantly, there is no need to use it to add data for now.
+    ENVOY_LOG(error, "golang filter calls addData when processing data is not supported, use "
+                     "`BufferInstance.Append` instead.");
+    return CAPIStatus::CAPIInvalidPhase;
+  }
+
+  Thread::LockGuard lock(mutex_);
+  if (has_destroyed_) {
+    ENVOY_LOG(debug, "golang filter has been destroyed");
+    return CAPIStatus::CAPIFilterIsDestroy;
+  }
+  if (!state.isProcessingInGo()) {
+    ENVOY_LOG(debug, "golang filter is not processing Go");
+    return CAPIStatus::CAPINotInGo;
+  }
+
+  if (state.isThreadSafe()) {
+    addDataInternal(state, data, is_streaming);
+    return CAPIStatus::CAPIOK;
+  }
+
+  auto weak_ptr = weak_from_this();
+  state.getDispatcher().post([this, weak_ptr, &state, data, is_streaming] {
+    if (!weak_ptr.expired() && !hasDestroyed()) {
+      addDataInternal(state, data, is_streaming);
```

Comment:
It's unsafe to use data pointer from Golang side, after returning to Golang.

Should copy for later using, i.e.
https://github.com/envoyproxy/envoy/blob/96b345ff93d6268165603cf3c49d0027c553c41a/contrib/golang/filters/http/source/golang_filter.cc#L681-L683

Reply:
@doujiang24 
Good catch! Fixed in new commit.